Prompt DQM for Jet Trigger (HLTJet)

physicist87 opened the pull request:
I updated Prompt DQM for HLT Jet Trigger.

Previous PR (https://github.com/cms-sw/cmssw/pull/19076 )was screwed up. 

I don't know how I recover the previous "PR", So I created new PR for Prompt DQM for HLT Jet Trigger.

physicist87 committed:
Update Prompt DQM for HLT Jet

physicist87 committed:
Update Prompt DQM for HLT Jet

cmsbuild commented:
A new Pull Request was created by  @physicist87 (Seungkyu Ha) for master.

It involves the following packages:

DQMOffline/Trigger


@cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@battibass, @mtosi, @jhgoh, @calderona, @HuguesBrun, @trocino, @rociovilar this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

mtosi commented:
tracked by #19142

mtosi commented on DQMOffline/Trigger/plugins/JetMonitor.cc:
```diff
@@ -0,0 +1,594 @@
+   }
+   hName = h_Name+"pT"+hSubN;
+   hTitle = h_Title+" pT " + hSubT;
+   bookME(Ibooker,a_me[0],hName,hTitle,met_binning_.nbins,met_binning_.xmin, met_binning_.xmax);
```
are you sure this is "met" ?!?!?

physicist87 replied:
I didn't changed the binning variable name. 
At here, I used met_binning, It's for Jet pT. 
If you want that I change this name, I will change it.

mtosi commented on DQMOffline/Trigger/plugins/JetMonitor.cc:
```diff
@@ -0,0 +1,594 @@
+  v_jetpt.clear();
+  v_jeteta.clear();
+  v_jetphi.clear();
+  std::vector<reco::PFJet> pfjets;
```
did you try edm::View<reco::Jet> ?

physicist87 replied:
I didn't try using the edm::View <reco::Jet>.
I focused on reducing copy-and-paste. 
Do you want me to use edm::View <reco::Jet>?

mtosi commented on DQMOffline/Trigger/python/JetMETPromptMonitoring_Client_cff.py:
```diff
@@ -0,0 +1,172 @@
+        "effic_pfjetEtaVsPhi_HEP18 'Jet efficiency vs #eta and #phi(HEP18); PF Jet #eta; #phi' pfjetEtaVsPhi_HEP18_numerator       pfjetEtaVsPhi_HEP18_denominator",
+        "effic_pfjetEtaVsPhi_HE_p 'Jet efficiency vs #eta and #phi(HE_p); PF Jet #eta; #phi' pfjetEtaVsPhi_HE_p_numerator       pfjetEtaVsPhi_HE_p_denominator",
+        "effic_pfjetEtaVsPhi_HE_m 'Jet efficiency vs #eta and #phi(HE_m); PF Jet #eta; #phi' pfjetEtaVsPhi_HE_m_numerator       pfjetEtaVsPhi_HE_m_denominator",
+    ),
```
why don't you add ratio between HEP17 and HEM17 etc ?

mtosi replied:
in addition, I see only efficiency, what about resolution and other observables simone pointed you ?

sehwook replied:
Dear Mia,
I am very grateful to you for your reviewing Seungkyu's code.
On behalf of him, I would like to know what ratio plot he has to put.
Do you mean the ratio of efficiency of HEP 17 to that of HEM 17 as a function of pT and phi?
In case of eta, HEP17 and HEM17 have different sign of eta's.
Do you want him to add it as a function of absolute eta?

Thank you,
Sehwook

sehwook replied:
Dear Mia,

For resolution, I know Simon draw the resolution with hltjet.pt()/genjet.pt().
In this case, I know he uses the prompt reco jet in data, so that I am not sure he could use genjet.
How can he define the resolution in this case?

As far as I know, we could draw the jet  pT or energy distribution and calculate rms/mean since the response of calorimeter to jets would not be Gaussian.
This will be the jet energy resolution in calorimetry.

If I missed something, please let me know.

Thank you,
Sehwook

mtosi commented:
what about 
HLT_AK8PFJet360_TrimMass30_v*
HLT_AK8PFHT750_TrimMass50_v*
HLT_AK8PFJet380_TrimMass30_v*
HLT_AK8PFHT800_TrimMass50_v*
HLT_AK8PFJet400_TrimMass30_v*
HLT_AK8PFHT850_TrimMass50_v*
HLT_AK8PFJet420_TrimMass30_v*
HLT_AK8PFHT900_TrimMass50_v*
?
are they going to be monitored by B2G @rappoccio @alefisico ?

mtosi commented:
yes, please

On Mon, Jun 12, 2017 at 11:09 AM, sehwook <notifications@github.com> wrote:

> *@sehwook* commented on this pull request.
> ------------------------------
>
> In DQMOffline/Trigger/python/JetMETPromptMonitoring_Client_cff.py
> <https://github.com/cms-sw/cmssw/pull/19178#discussion_r121342828>:
>
> > +
> +        "effic_pfjetpT_HF          'Jet pT turnON (HF);            PFJet(pT) [GeV]; efficiency'     pfjetpT_HF_numerator          pfjetpT_HF_denominator",
> +        "effic_pfjetpT_HF_pTThresh 'Jet pT turnON (HF);            PFJet(pT) [GeV]; efficiency'     pfjetpT_pTThresh_HF_numerator pfjetpT_pTThresh_HF_denominator",
> +        "effic_pfjetPhi_HF       'Jet efficiency vs #phi (HF); PF Jet phi [rad]; efficiency' pfjetPhi_HF_numerator       pfjetPhi_HF_denominator",
> +        "effic_pfjetEta_HF       'Jet efficiency vs #eta (HF); PF Jet phi [rad]; efficiency' pfjetEta_HF_numerator       pfjetEta_HF_denominator",
> +        ## 2D Eff
> +        "effic_pfjetEtaVsPhi       'Jet efficiency vs #eta and #phi; PF Jet #eta; #phi' pfjetEtaVsPhi_numerator       pfjetEtaVsPhi_denominator",
> +        "effic_pfjetEtaVsPhi_HB    'Jet efficiency vs #eta and #phi(HB); PF Jet #eta; #phi' pfjetEtaVsPhi_HB_numerator       pfjetEtaVsPhi_HB_denominator",
> +        "effic_pfjetEtaVsPhi_HE    'Jet efficiency vs #eta and #phi(HE); PF Jet #eta; #phi' pfjetEtaVsPhi_HE_numerator       pfjetEtaVsPhi_HE_denominator",
> +        "effic_pfjetEtaVsPhi_HF    'Jet efficiency vs #eta and #phi(HF); PF Jet #eta; #phi' pfjetEtaVsPhi_HF_numerator       pfjetEtaVsPhi_HF_denominator",
> +        "effic_pfjetEtaVsPhi_HEM17 'Jet efficiency vs #eta and #phi(HEM17); PF Jet #eta; #phi' pfjetEtaVsPhi_HEM17_numerator       pfjetEtaVsPhi_HEM17_denominator",
> +        "effic_pfjetEtaVsPhi_HEP17 'Jet efficiency vs #eta and #phi(HEP17); PF Jet #eta; #phi' pfjetEtaVsPhi_HEP17_numerator       pfjetEtaVsPhi_HEP17_denominator",
> +        "effic_pfjetEtaVsPhi_HEP18 'Jet efficiency vs #eta and #phi(HEP18); PF Jet #eta; #phi' pfjetEtaVsPhi_HEP18_numerator       pfjetEtaVsPhi_HEP18_denominator",
> +        "effic_pfjetEtaVsPhi_HE_p 'Jet efficiency vs #eta and #phi(HE_p); PF Jet #eta; #phi' pfjetEtaVsPhi_HE_p_numerator       pfjetEtaVsPhi_HE_p_denominator",
> +        "effic_pfjetEtaVsPhi_HE_m 'Jet efficiency vs #eta and #phi(HE_m); PF Jet #eta; #phi' pfjetEtaVsPhi_HE_m_numerator       pfjetEtaVsPhi_HE_m_denominator",
> +    ),
>
> Dear Mia,
> I am very grateful to you for your reviewing Seungkyu's code.
> On behalf of him, I would like to know what ratio plot he has to put.
> Do you mean the ratio of efficiency of HEP 17 to that of HEM 17 as a
> function of pT and phi?
> In case of eta, HEP17 and HEM17 have different sign of eta's.
> Do you want him to add it as a function of absolute eta?
>
> Thank you,
> Sehwook
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19178#discussion_r121342828>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt584jYbfbOIQbkXLR_NPa6L7lYzRIrks5sDQBggaJpZM4N1lmF>
> .
>

physicist87 commented:
what about
HLT_AK8PFJet360_TrimMass30_v*
HLT_AK8PFHT750_TrimMass50_v*
HLT_AK8PFJet380_TrimMass30_v*
HLT_AK8PFHT800_TrimMass50_v*
HLT_AK8PFJet400_TrimMass30_v*
HLT_AK8PFHT850_TrimMass50_v*
HLT_AK8PFJet420_TrimMass30_v*
HLT_AK8PFHT900_TrimMass50_v*
?
are they going to be monitored by B2G @rappoccio @alefisico ?

-> I don't know they are monitoring these triggers.
Should I add these triggers ??

mtosi commented:
I think it would be better
  mia

On Mon, Jun 12, 2017 at 11:29 AM, Seungkyu Ha <notifications@github.com>
wrote:

> *@physicist87* commented on this pull request.
> ------------------------------
>
> In DQMOffline/Trigger/plugins/JetMonitor.cc
> <https://github.com/cms-sw/cmssw/pull/19178#discussion_r121346900>:
>
> > +#include "Geometry/Records/interface/TrackerTopologyRcd.h"
> +#include "DataFormats/Math/interface/deltaR.h" // For Delta R
> +void JetMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)  {
> +  // Filter out events if Trigger Filtering is requested
> +  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
> +  edm::Handle<reco::PFJetCollection> pfjetHandle;
> +  iEvent.getByToken( pfjetToken_, pfjetHandle );
> +
> +  edm::Handle<reco::CaloJetCollection> calojetHandle;
> +  iEvent.getByToken( calojetToken_, calojetHandle );
> +
> +  int ls = iEvent.id().luminosityBlock();
> +  v_jetpt.clear();
> +  v_jeteta.clear();
> +  v_jetphi.clear();
> +  std::vector<reco::PFJet> pfjets;
>
> I didn't try using the edm::View reco::Jet.
> I focused on reducing copy-and-paste.
> Do you want me to use edm::View reco::Jet?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19178#discussion_r121346900>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt58zZPIbvMDEqf5cG76M6xoE93u5gcks5sDQT0gaJpZM4N1lmF>
> .
>

mtosi commented:
ah !
I did not know simone was using GEN info :(
sorry for the confusion
if you want you can make use of the offline quantity ...

On Mon, Jun 12, 2017 at 11:32 AM, sehwook <notifications@github.com> wrote:

> *@sehwook* commented on this pull request.
> ------------------------------
>
> In DQMOffline/Trigger/python/JetMETPromptMonitoring_Client_cff.py
> <https://github.com/cms-sw/cmssw/pull/19178#discussion_r121347523>:
>
> > +
> +        "effic_pfjetpT_HF          'Jet pT turnON (HF);            PFJet(pT) [GeV]; efficiency'     pfjetpT_HF_numerator          pfjetpT_HF_denominator",
> +        "effic_pfjetpT_HF_pTThresh 'Jet pT turnON (HF);            PFJet(pT) [GeV]; efficiency'     pfjetpT_pTThresh_HF_numerator pfjetpT_pTThresh_HF_denominator",
> +        "effic_pfjetPhi_HF       'Jet efficiency vs #phi (HF); PF Jet phi [rad]; efficiency' pfjetPhi_HF_numerator       pfjetPhi_HF_denominator",
> +        "effic_pfjetEta_HF       'Jet efficiency vs #eta (HF); PF Jet phi [rad]; efficiency' pfjetEta_HF_numerator       pfjetEta_HF_denominator",
> +        ## 2D Eff
> +        "effic_pfjetEtaVsPhi       'Jet efficiency vs #eta and #phi; PF Jet #eta; #phi' pfjetEtaVsPhi_numerator       pfjetEtaVsPhi_denominator",
> +        "effic_pfjetEtaVsPhi_HB    'Jet efficiency vs #eta and #phi(HB); PF Jet #eta; #phi' pfjetEtaVsPhi_HB_numerator       pfjetEtaVsPhi_HB_denominator",
> +        "effic_pfjetEtaVsPhi_HE    'Jet efficiency vs #eta and #phi(HE); PF Jet #eta; #phi' pfjetEtaVsPhi_HE_numerator       pfjetEtaVsPhi_HE_denominator",
> +        "effic_pfjetEtaVsPhi_HF    'Jet efficiency vs #eta and #phi(HF); PF Jet #eta; #phi' pfjetEtaVsPhi_HF_numerator       pfjetEtaVsPhi_HF_denominator",
> +        "effic_pfjetEtaVsPhi_HEM17 'Jet efficiency vs #eta and #phi(HEM17); PF Jet #eta; #phi' pfjetEtaVsPhi_HEM17_numerator       pfjetEtaVsPhi_HEM17_denominator",
> +        "effic_pfjetEtaVsPhi_HEP17 'Jet efficiency vs #eta and #phi(HEP17); PF Jet #eta; #phi' pfjetEtaVsPhi_HEP17_numerator       pfjetEtaVsPhi_HEP17_denominator",
> +        "effic_pfjetEtaVsPhi_HEP18 'Jet efficiency vs #eta and #phi(HEP18); PF Jet #eta; #phi' pfjetEtaVsPhi_HEP18_numerator       pfjetEtaVsPhi_HEP18_denominator",
> +        "effic_pfjetEtaVsPhi_HE_p 'Jet efficiency vs #eta and #phi(HE_p); PF Jet #eta; #phi' pfjetEtaVsPhi_HE_p_numerator       pfjetEtaVsPhi_HE_p_denominator",
> +        "effic_pfjetEtaVsPhi_HE_m 'Jet efficiency vs #eta and #phi(HE_m); PF Jet #eta; #phi' pfjetEtaVsPhi_HE_m_numerator       pfjetEtaVsPhi_HE_m_denominator",
> +    ),
>
> Dear Mia,
>
> For resolution, I know Simon draw the resolution with hltjet.pt()/
> genjet.pt().
> In this case, I know he uses the prompt reco jet in data, so that I am not
> sure he could use genjet.
> How can he define the resolution in this case?
>
> As far as I know, we could draw the jet pT or energy distribution and
> calculate rms/mean since the response of calorimeter to jets would not be
> Gaussian.
> This will be the jet energy resolution in calorimetry.
>
> If I missed something, please let me know.
>
> Thank you,
> Sehwook
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19178#discussion_r121347523>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt58wjiPUaHL5D-LqdrLB61NoVtt-qyks5sDQW6gaJpZM4N1lmF>
> .
>

rappoccio commented:
@mtosi yes, @UBParker is implementing these.

dmitrijus commented on DQMOffline/Trigger/plugins/JetMonitor.cc:
```diff
@@ -0,0 +1,594 @@
+      v_me[6].numerator->Fill(eta_,pt_);// eta vs phi
+   }
+   else {
+   std::cout << "CHECK OUT denu option in FillME !!!"<< std::endl;
```
use edm::LogWarning

physicist87 committed:
Update To use edm::View<reco::Jet>

physicist87 committed:
Update To use edm::View<reco::Jet>

physicist87 committed:
Ratio Plot(HEP17/HEM17) for eta phi pT

cmsbuild commented:
Pull request #19178 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

physicist87 commented:
1. I updated Ratio plots (HEP17/HEM17) for eta, phi and pT. 

To do this thing, I made JetPomptDQMPostProcessor.cc which can control Histograms information at Harvest step. 

2. I changed std::out to edm::WarningLog. 

3. In JetMETPromptMonitoring_Client_cff.py, I used wrong name to make efficiency plot for eta and phi.
I fixed it.

4. I use edm::view<reco::Jet> instead of using <reco::PFJet> and <reco::CaloJet>

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

cmsbuild commented:
-1

Tested at: 97f4f261786b52e3e4ba765036fbf396fde98b5a

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19178/20817/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
140.53 step2
<pre>runTheMatrix-results/140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI/step2_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

physicist87 committed:
To avoid the case of non ak4pfjet, bug fix

cmsbuild commented:
Pull request #19178 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

physicist87 commented:
Bug fix.  @cmsbuild could you please test again ?

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

cmsbuild commented:
+1
Tested at: aaa57c09bb28cdfa75864e1c6dbd7945efd3acbd
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19178/20880/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5ad2ca330590b7d14a64a053f7f4350de7357460
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19178/20880/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19178/20880/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19178/20880/summary.html

Comparison Summary:
 - You potentially added 5004 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1802916
 - DQMHistoTests: Total failures: 29337
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1773413
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

mtosi commented:
I would suggest/request to
1. make JetPomptDQMPostProcessor more generic, in order to allow other users to make use of it
2. check w/ DQM core ( @dmitrijus, @vanbesien) how to implement this functionality in the already available DQMGenericClient

physicist87 commented:
@mtosi 
Dear Mia,

I'm trying to fix JetPomptDQMPostProcessor to be more generic. 

But I have some questions.

I would like to know how I make JetPomptDQMPostProcessor more generic.

Do you mean I keep JetPomptDQMPostProcessor, and just modify this code to control the histograms in the python configuration files ? 

If not, Do want me to modify DQMGenericClient to do make ratio plots?

Thank you,
Seungkyu

mtosi commented:
ciao

please, make it configurable via python as now
but I would like some feedback from DQM team

  mia

On Wed, Jun 28, 2017 at 6:52 AM, Seungkyu Ha <notifications@github.com>
wrote:

> @mtosi <https://github.com/mtosi>
> Dear Mia,
>
> I'm trying to fix JetPomptDQMPostProcessor to be more generic.
>
> But I have some questions.
>
> I would like to know how I make JetPomptDQMPostProcessor more generic.
>
> Do you mean I keep JetPomptDQMPostProcessor, and just modify this code to
> control the histograms in the python configuration files ?
>
> If not, Do want me to modify DQMGenericClient to do make ratio plots?
>
> Thank you,
> Seungkyu
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19178#issuecomment-311556022>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt580pDpGwcA2g59hlA6egYFmjDVJ3Oks5sIdwUgaJpZM4N1lmF>
> .
>

dmitrijus commented:
Hi,

I don't know what exact functionality you are talking about, if it is just "ratio" and nothing HLT specific,
feel free to modify DQMGenericCliient. I will check these changes anyway.

Dmitrijus

physicist87 commented:
@mtosi @dmitrijus 

Dear Mia and Dmitrijus

I see. I'm trying to modify DQMGenericClient to make ratio plot of efficiency. 

If I finish this work, I will update PR. 

Thank you,
Seungkyu

physicist87 committed:
fix uncertainty for simpleratio in DQMGenericClient

physicist87 committed:
There is no need to use JetPomptDQMPostProcessor, because we will use DQMGenericClient

physicist87 committed:
Update JetMETPromptMonitoring_Client_cff, To use DQMGenericClient instead of JetPomptDQMPostProcessor

cmsbuild commented:
Pull request #19178 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

physicist87 commented:
@mtosi  @dmitrijus 
Dear Mia and Dmitrijus,

As I undestand DQMGenericClient code, we are using the generic_eff function to make efficiency plot. 

Unfortunately, for option of simpleratio, with out profile option, the uncertainty of efficiency (or ratio) is wrong. [1]

So I fixed this part and modified our python files to use DQMGenericClient. 

If you go to this link [2], you can see the result with modified DQMGenericClient. 

Best regards,
Seungkyu

[1] 
https://github.com/cms-sw/cmssw/blob/master/DQMServices/ClientConfig/plugins/DQMGenericClient.cc#L963

[2]
https://cernbox.cern.ch/index.php/s/hcoJU4OkpdLHNJk

mtosi commented:
thanks !
@dmitrijus are we (better, you) fine w/ this ?
can we proceed w/ the integration ?

@cmsbuild, please test

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 933b1cf0361066f5307b7fcbb3b39c94c31f3699
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19178/21085/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/b1580cdb9b247f8d4a93ce44ec73c8c8213a595e
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19178/21085/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19178/21085/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19178/21085/summary.html

Comparison Summary:
 - You potentially added 5002 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1756063
 - DQMHistoTests: Total failures: 29337
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1726560
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

davidlange6 commented:
@physicist87  while waiting for your DQM review - there is now a base class appropriate for your new modules available at https://github.com/cms-sw/cmssw/blob/master/DQMOffline/Trigger/plugins/TriggerDQMBase.h. Please have a look and adapt your module. In addition to removing code duplication, you will have addressed a number of the "usual" comments on PRs like this one likely resulting in a faster review.

physicist87 commented:
@davidlange6 Thank you for your suggestion. I'm trying to follow your suggestion. If I finish this work, I  will update this PR.
Thank you,
Seungkyu

dmitrijus commented on DQMServices/ClientConfig/plugins/DQMGenericClient.cc:
```diff
@@ -960,7 +960,11 @@ void DQMGenericClient::generic_eff (TH1* denom, TH1* numer, MonitorElement* effi
+          float numerErr = numer->GetBinError(globalBinNum);
+          float denomErr = denom->GetBinError(globalBinNum);
+          float denomsq = denomVal*denomVal;
+          errVal = denomVal ? sqrt( pow( 1.f/denomVal*numerErr,2.0) + pow(numerVal/denomsq*denomErr,2)  ): 0;
```
This changes the behaviour of all the modules which use this "generic_eff". But I guess you know what you are doing :)

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

mtosi commented:
can you please point to the back-port in 92x, please ?
it is needed in order to have the code run in the upcoming prompt reco ...
thanks !

physicist87 commented:
@mtosi 
Dear Mia, 
Here is our back-port PR https://github.com/cms-sw/cmssw/pull/19603

physicist87 committed:
the ME initialization into Struct define and modify nbins unsigned int, and remove not used function and variables

physicist87 committed:
the ME initialization into Struct define and modify nbins unsigned int, and remove not used function and variables

physicist87 committed:
nbins changed into unsigned int

physicist87 committed:
nbins changed into unsigned int

cmsbuild commented:
Pull request #19178 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @vanbesien, @davidlange6 can you please check and sign again.

physicist87 commented:
@mtosi @dmitrijus @davidlange6 
Dear all,

I updated this PR to handle David's comments at https://github.com/cms-sw/cmssw/pull/19372 PR. 
If you see #19372 PR, David left some comment about my codes. 
So I modified my codes.
These are list of what I modified. 
1. I changed nbins as unsigned int 
2. Remove not used functions and variables. 
3. Move ME initialization into Struct definition. 
4. Changed the metPset as jetPSet "because these are for Jet Trigger."

And also Back-port PR is updated. #19603 

However, I didn't use TriggerDQMBase. 

@mtosi  I'm still confused. I looked into amagnan'PR to follow David's comment. 
As you know, amagnan made ObjMonitor and JetDQM to use TriggerDQMBase. 
As I understand these code, ObjMonitor is main code and JetDQM is for DQM functionality. 
Do you also want me to use these format ? 
If you want to do that, I need to modify most of my JetMonitor code. 

If you don't mind, could you tell me your opinion ?

Best regards,
Seungkyu

mtosi commented:
ciao

moving to a more general/modular approach would be great, indeed
but I think you should have the DQM code ready to be integrated asap !

I would suggest to
1. try to have a reasonable version of your monitoring integrated asap
2. in the upcoming weeks, update your code in order to make it as
general/modular as possible

do you agree ?
  mia

On Fri, Jul 7, 2017 at 8:43 AM, Seungkyu Ha <notifications@github.com>
wrote:

> @mtosi <https://github.com/mtosi> @dmitrijus
> <https://github.com/dmitrijus> @davidlange6
> <https://github.com/davidlange6>
> Dear all,
>
> I updated this PR to handle David's comments at #19372
> <https://github.com/cms-sw/cmssw/pull/19372> PR.
> If you see #19372 <https://github.com/cms-sw/cmssw/pull/19372> PR, David
> left some comment about my codes.
> So I modified my codes.
> These are list of what I modified.
>
>    1. I changed nbins as unsigned int
>    2. Remove not used functions and variables.
>    3. Move ME initialization into Struct definition.
>    4. Changed the metPset as jetPSet "because these are for Jet Trigger."
>
> And also Back-port PR is updated. #19603
> <https://github.com/cms-sw/cmssw/pull/19603>
>
> However, I didn't use TriggerDQMBase.
>
> @mtosi <https://github.com/mtosi> I'm still confused. I looked into
> amagnan'PR to follow David's comment.
> As you know, amagnan made ObjMonitor and JetDQM to use TriggerDQMBase.
> As I understand these code, ObjMonitor is main code and JetDQM is for DQM
> functionality.
> Do you also want me to use these format ?
> If you want to do that, I need to modify most of my JetMonitor code.
>
> If you don't mind, could you tell me your opinion ?
>
> Best regards,
> Seungkyu
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19178#issuecomment-313600772>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt580y5Z8mMJOzFNRvGzrjJrFkOLbQkks5sLdOLgaJpZM4N1lmF>
> .
>

physicist87 commented:
@mtosi 
I agree with you. 
I'm trying to adapt TriggerDQMBase code into my codes. 

Thank you,
Seungkyu

physicist87 commented:
@mtosi  The back port of this pr is #19603 .

I think #19603 is working on 92x and reasonable version without general class. 

After #19603 is merged, I will update this pr with general class.

Best regards,
Seungkyu

fwyzard commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

fwyzard commented:
Please merge #19684 which includes this and fixes conflicts.

cmsbuild commented:
+1
Tested at: 055628e4c702a2fa6f384309ab28c05d08458e19
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19178/21366/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/92ee79880f3dc245397a00e6acced67d57d310cc
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19178/21366/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19178/21366/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19178/21366/summary.html

Comparison Summary:
 - You potentially added 4999 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 2012315
 - DQMHistoTests: Total failures: 100
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2012049
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 93 log files, 14 edm output root files, 23 DQM output files

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar
